### PR TITLE
Some work on adding Android support

### PR DESCRIPTION
**Pull request by jessicafalk**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR enables building the Unreal GDK for Android, downloads the necessary packages and fixes most of the build issues.. 
A second PR will be added later that will resolve any outstanding build issues and enable Android support for goo.

PR can be read in commit order

#### Release note
Will add a release note in the next PR which will contain a Worker SDK upgrade and make it possible to build & run Android application.

#### Tests
This is all manually tested and I've been writing documentation on how to use it.

#### Documentation
documentation is in progress

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStaticComponentView.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ void USpatialStaticComponentView::OnAddComponent(const Worker_AddComponentOp& Op
 		// Component is not hand written, but we still want to know the existence of it on this entity.
 		Data = nullptr;
 	}
-	EntityComponentMap.FindOrAdd(Op.entity_id).FindOrAdd(Op.data.component_id) = std::move(Data);
+	EntityComponentMap.FindOrAdd(Op.entity_id).FindOrAdd(Op.data.component_id) = MoveTemp(Data);
```
Comment (jayprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStaticComponentView.cpp]: I'd encourage this to be separated out into a different PR. This is 1) a case where a stricter compiler complained about a missing include (e.g. &lt;algorithm&gt;) 2) a case where a code style wasn't followed, .. std::move and most stl isn't prevelant in the Unreal codebase. As a separate PR hopefully it stands out as an example for others about knowing when code doesn't match the style.

Comment (jessicafalk) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStaticComponentView.cpp]: double checked who made that change. it was @neprune  and @danielimprobable sometime in 2018, so a pretty long time ago.
This is the only occurrence of using a STL function in the Unreal GDK, so I'm not too worried about it happening again.

File: SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs [hunk 2]
```diff
@@ -89,6 +89,18 @@ public SpatialGDK(ReadOnlyTargetRules Target) : base(Target)
...
+        {
+            // Android behaves slightly differently and requires an additional XML file to be able to copy the libraries over.
+            PublicLibraryPaths.Add(Path.Combine(WorkerLibraryDir, "arm64-v8a"));
+            PublicLibraryPaths.Add(Path.Combine(WorkerLibraryDir, "armeabi-v7a"));
+            PublicLibraryPaths.Add(Path.Combine(WorkerLibraryDir, "x86_64"));
+            PublicAdditionalLibraries.Add("improbable_worker");
+            string PluginPath = Utils.MakePathRelativeTo(ModuleDirectory, Target.RelativeEnginePath);
+            AdditionalPropertiesForReceipt.Add(new ReceiptProperty("AndroidPlugin", Path.Combine(PluginPath, "SpatialGDK_APL.xml")));
+            PublicDefinitions.Add("TRACE_LIB_ACTIVE=0");
+            return;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs]: This return is slightly painful, anyway android can be re-worked so the flows are more compatible?

File: Setup.bat [hunk 3]
```diff
@@ -99,6 +100,9 @@ call :MarkStartOfBlock "Retrieve dependencies"
     spatial package retrieve worker_sdk      c-dynamic-x86_64-vc140_md-win32            %PINNED_CORE_SDK_VERSION%   %DOMAIN_ENVIRONMENT_VAR%   "%CORE_SDK_DIR%\worker_sdk\c-dynamic-x86_64-vc140_md-win32.zip"
     spatial package retrieve worker_sdk      c-dynamic-x86_64-gcc510-linux              %PINNED_CORE_SDK_VERSION%   %DOMAIN_ENVIRONMENT_VAR%   "%CORE_SDK_DIR%\worker_sdk\c-dynamic-x86_64-gcc510-linux.zip"
     spatial package retrieve worker_sdk      c-static-fullylinked-arm-clang-ios         %PINNED_CORE_SDK_VERSION%   %DOMAIN_ENVIRONMENT_VAR%   "%CORE_SDK_DIR%\worker_sdk\c-static-fullylinked-arm-clang-ios.zip"
+    spatial package retrieve worker_sdk      c-dynamic-arm64v8a-clang_ndk16b-android    %PINNED_CORE_SDK_VERSION%   %DOMAIN_ENVIRONMENT_VAR%   "%CORE_SDK_DIR%\worker_sdk\c-dynamic-arm64v8a-clang_ndk16b-android.zip"
```
Comment (m-samiec) [inline Setup.bat]: Any idea how big these packages are? Do we want to default include them?

Comment (jessicafalk) [reply Setup.bat]: Just checked. They are all between 5-6 MB big.  if you prefer, we can have a mobile flag or something similar that will download it if needed. This would probably be useful for the iOS package that is pretty big and has caused timeouts before.

File: SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs (hunk 2 above)
Comment (jessicafalk) [reply SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs]: will take another look

File: Setup.bat (hunk 3 above)
Comment (m-samiec) [reply Setup.bat]: Flag sounds sensible. cc: @ernopp

Comment (jessicafalk) [reply Setup.bat]: added a --mobile flag for both Setup scripts.
